pyoptsparse/pyOpt_history.py

```diff
@@ -433,7 +431,7 @@ def _processIterDict(self, d, scale=False):
         These are all "flat" dictionaries, with simple key:value pairs.
         """
         conDict = {}
-        for con in self.optProb.constraints.keys():
+        for con in list(self.optProb.constraints.keys()):
             # linear constraints are not stored in funcs
```

Comment:
Now that you explain it the bug is evident, but what kind of issues was it leading to? how did you catch it?

Reply:
You will get an error saying `OrderedDict mutated during iteration`